models/repo_transfer.go

```diff
@@ -325,6 +325,33 @@ func TransferOwnership(doer *User, newOwnerName string, repo *Repository) (err e
 		}
 	}
 
+	// Delete labels that belong to the old organization and comments that added these labels
+	if oldOwner.IsOrganization() {
+		if _, err := sess.Exec(`DELETE FROM issue_label WHERE issue_label.id IN (
+			SELECT il_too.id FROM (
+				SELECT il_too_too.id
+					FROM issue_label AS il_too_too
+						INNER JOIN label ON il_too_too.id = label.id
+						INNER JOIN issue on issue.id = il_too_too.issue_id
+					WHERE
+						issue.repo_id = ? AND (issue.repo_id != label.repo_id OR (label.repo_id = 0 AND label.org_id != ?))
+		) AS il_too )`, repo.ID, newOwner.ID); err != nil {
+			return fmt.Errorf("Unable to remove old org labels: %v", err)
```

Comment:
I'm not sure why we need to join issues in here,
this looks like a try to fix already inconsistent db things on the fly while transferring :/

Reply:
yup - because we're still not sure why this is happening and it's a good chance to clean things up. Consistency should be checked when we can.

Reply:
@zeripath can you simplify this in your followup and run migration to clean up all and for once istead?

I'll lgtm it for now but find it rather ... not the right place to see a fix

Reply:
We need to join `issue` because otherwise you cannot get all the `issue_labels` for a particular `repo.id`.

I take it you mean you want the additional consistency check removed?

Reply:
given the fact that we still have to join issue I'm not certain I understand completely the benefit of removing this.

Reply:
I'm fine ... still think this is a monster but let it run